crates/uniffi-bindgen-react-native/src/bindings/react_native/gen_typescript/templates/EnumTemplate.ts

```diff
@@ -42,104 +42,44 @@ const {{ ffi_converter_name }} = (() => {
 })();
 
 {% else %}
-/// Typescript doesn't have the concept of enums with variant properties,
-/// so we need to be a little creative here.
-///
-/// For a Rust enum:
-/// ```rs
-/// enum FilePath {
-///   Local { path: String }
-///   Remote { host: String, path: String }
-/// }
-/// ```
-/// Currently:
-/// ```ts
-/// class FilePath {
-///   static Local = class Local extends FilePath { constructor(public members: { path: string })} {}
-///   static Remote = class Remote extends FilePath { constructor(public members: { host: string, path: string })} {}
-/// }
-/// ```
-///
-/// This gives nice construction properties:
-/// ```ts
-/// const path: FilePath = new FilePath.Local({ path: "/" });
-/// ```
-/// but not very nice pattern matching.
-///
-/// To help with switch statements, a companion enum (not ideal) called `FilePathKind` is generated:
-///
-/// ```ts
-/// enum FilePathKind {
-///   LOCAL,
-///   REMOTE
-/// }
-/// ```
-///
-/// so:
-/// ```ts
-/// switch path.kind {
-///     case FilePathKind.LOCAL:
-///         console.log("It's a local file path");
-///         break;
-///     case FilePathKind.REMOTE:
-///         console.log("It's a remote file path");
-///         break;
-/// }
-/// ```
-///
-///
+
+// Enum: {{ type_name }}
 {%- let kind_type_name = format!("{type_name}Kind") %}
 export enum {{ kind_type_name }} {
     {%- for variant in e.variants() %}
-    {{ variant|variant_name }} = {{ loop.index0 }}
+    {{ variant|variant_name }} = "{{ variant.name() }}"
     {%- if !loop.last %},{% endif -%}
     {% endfor %}
 }
 
 {%- call ts::docstring(e, 0) %}
-export abstract class {{ type_name }} {
-    protected constructor(public kind: {{ kind_type_name }}) {}
-
-    {%-   for variant in e.variants() %}
-    {%-    call ts::docstring(variant, 4) %}
-    {%-    let var_name = variant.name()|class_name(ci) %}
-    static {{ var_name }} = class {{ var_name }} extends {{ type_name }} {
-        constructor(
-        {%- if !variant.fields().is_empty() %}
-            public members: {
-                {%- for field in variant.fields() %}
-                {% call ts::field_name(field, loop.index) %}: {{ field|type_name(ci) }}
-                {%- if loop.last -%}{%- else -%},{%- endif -%}{% endfor %}
-            }
-        {%- endif %}
-        ) { super({{kind_type_name}}.{{ variant|variant_name }}); }
-
-        asJSON(): any {
-            {%- if !variant.fields().is_empty() %}
-            return { ["{{ type_name }}.{{ var_name }}"]: this.members };
-            {%- else %}
-            return "{{ type_name }}.{{ var_name }}";
-            {%- endif %}
-        }
-    }
-    {%- endfor %}
-}
+export type {{ type_name }} = {# space #}
+{%- for variant in e.variants() %}
+{%-   call ts::docstring(variant, 4) %}
+    { kind: {{ kind_type_name }}.{{ variant|variant_name }}
+    {%- if !variant.fields().is_empty() %}, value: {
```

Comment:
I believe that technically in typescript, you separate type definitions with `;` instead of `,`.  It accepts both because 'javascript' but I'm pretty sure this should be `; value:`

Reply:
wow. TIL.